doctests/dt-streams.js

```diff
@@ -0,0 +1,340 @@
+// EXAMPLE: streams_tutorial
+// HIDE_START
+import assert from 'assert';
+import {
+  createClient
+} from 'redis';
+
+const client = await createClient();
+await client.connect();
+// HIDE_END
+// REMOVE_START
+await client.flushDb();
+// REMOVE_END
+
+// STEP_START xAdd
+const res1 = await client.xAdd(
+  'race:france', '*', {
+    'rider': 'Castilla',
+    'speed': '30.2',
+    'position': '1',
+    'location_id': '1'
+  }
+);
+console.log(res1); // >>> 1700073067968-0
+
+const res2 = await client.xAdd(
+  'race:france', '*', {
+    'rider': 'Norem',
+    'speed': '28.8',
+    'position': '3',
+    'location_id': '1'
+  },
+);
+console.log(res2); // >>> 1692629594113-0
+
+const res3 = await client.xAdd(
+  'race:france', '*', {
+    'rider': 'Prickett',
+    'speed': '29.7',
+    'position': '2',
+    'location_id': '1'
+  },
+);
+console.log(res3); // >>> 1692629613374-0
+// STEP_END
+
+// REMOVE_START
+assert.equal(await client.xLen('race:france'), 3);
+// REMOVE_END
+
+// STEP_START xRange
+const res4 = await client.xRange('race:france', '1691765278160-0', '+', 2);
+console.log(res4); // >>> [('1692629576966-0', {'rider': 'Castilla', 'speed': '30.2', 'position': '1', 'location_id': '1'}), ('1692629594113-0', {'rider': 'Norem', 'speed': '28.8', 'position': '3', 'location_id': '1'})]
+// STEP_END
+
+// STEP_START xread_block
+const res5 = await client.xRead({
+  key: 'race:france',
+  id: '0-0'
+}, {
+  count: 100,
+  block: 300
+});
+console.log(res5); // >>> [['race:france', [('1692629576966-0', {'rider': 'Castilla', 'speed': '30.2', 'position': '1', 'location_id': '1'}), ('1692629594113-0', {'rider': 'Norem', 'speed': '28.8', 'position': '3', 'location_id': '1'}), ('1692629613374-0', {'rider': 'Prickett', 'speed': '29.7', 'position': '2', 'location_id': '1'})]]]
```

Comment:
note that these id values dont correspond to the comments above, perhaps there should be more consistency in how we refer to variable" settings (ala IDs)